libwarpaffine/sliceswriter/SlicesWriterTbb.cpp

```diff
@@ -75,6 +79,35 @@ void CziSlicesWriterTbb::WriteWorker()
 
             add_subblock_info.ptrData = sub_block_write_info.add_slice_info.subblock_raw_data->GetPtr();
             add_subblock_info.dataSize = sub_block_write_info.add_slice_info.subblock_raw_data->GetSizeOfData();
+
+            if (this->use_acquisition_tiles_) 
+            {
+                int z;
+                sub_block_write_info.add_slice_info.coordinate.TryGetPosition(libCZI::DimensionIndex::Z, &z);
+                auto guid = this->CreateRetilingIdWithZandSlice(z, sub_block_write_info.add_slice_info.slice_id);
+
+                std::ostringstream oss;
+                oss << "<METADATA><Tags><RetilingId>"
+                    << std::hex << std::uppercase
+                    << std::setw(8) << std::setfill('0') << guid.Data1 << '-'
+                    << std::setw(4) << std::setfill('0') << guid.Data2 << '-'
+                    << std::setw(4) << std::setfill('0') << guid.Data3 << '-'
+                    << std::setw(2) << static_cast<int>(guid.Data4[0])
```

Comment:
is this cast necessary?

Reply:
yes, it matters. see https://glot.io/snippets/gxi89kqr9p

Reply:
OK, I see. What about a cast to uint32_t - would comfort my "be wary of casts between unsigned und signed types"-paranoia. And seems to work ok (tested in your playground).